internal/etw/fieldopt.go

```diff
@@ -28,6 +60,313 @@ func StringArray(name string, values []string) FieldOpt {
 	}
 }
 
+// IntField adds a single int field to the event.
+func IntField(name string, value int) FieldOpt {
+	switch unsafe.Sizeof(value) {
+	case 4:
+		return Int32Field(name, int32(value))
+	case 8:
+		return Int64Field(name, int64(value))
+	default:
+		panic("Unsupported int size")
+	}
+}
+
+// IntArray adds an array of int to the event.
+func IntArray(name string, values []int) FieldOpt {
+	inType := InTypeNull
+	var writeItem func(*EventData, int)
+	switch unsafe.Sizeof(values[0]) {

```

Comment:
This will panic if len 0

Reply:
This actually seems to work, due to unsafe.Sizeof being a compile-time constant expression. The tests I'm adding to log empty slices of int, uint, and uintptr all pass.